Assignment 5.Rmd

```diff
@@ -40,38 +50,84 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Take a look at our outcome variable "score"
 ```{r}
+
 hist(D1$score)
+
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
+
 D1$advice <- ifelse(D1$score <=0.4, "intervene", ifelse(D1$score > 0.4 & D1$score <=0.8, "monitor", "no action"))
+
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
+
 score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, D1)
 ```
 
 #Plot tree
 ```{r}
+
 plot(score_ctree)
+
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+######
+I would look at students that have asked for more than 12 hints or students that have gotten less than 63% of previous problems correct.
+
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice for new students based on tree generated from old students
 
 D2$prediction <- predict(score_ctree, D2)
 
 ``` 
+
+```{r}
+
+pred_ctree <- ctree(factor(prediction) ~ prior_prob_count + prior_percent_correct + hints, D2)
+
+plot(pred_ctree)
+
+sum(D2$prediction=="monitor")
+
+```
+
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+
+```{r}
+
+geom_bar(aes(D1$advice, D1$action), D1)
+
+
+d <- ggplot(D1, aes(advice, as.factor(action)))
+
+
+d + geom_bin2d(bins=3) 
+
+
+
+
++ geom_text(stat='count', aes(label=..count..), vjust=-1)
+
+
+
+, labels = count.fields(advice, action)
+
+```
+
```

Comment:
Can you provide an explanation of your code? I'm not sure how the visualizations answer the question?